src/core/utils/hooks/useTheme.ts

```diff
@@ -40,16 +40,33 @@ export const useTheme = (
       return;
     }
 
+    const prefersDarkMediaQuery = getWindow()?.matchMedia?.(
+      '(prefers-color-scheme: dark)',
+    );
+
+    const addOSTheme = ({
```

Comment:
You need to memoize this function because every rerender it will have different reference and it won't be removed from listeners.